app/controllers/documents_controller.rb

```diff
@@ -3,44 +3,29 @@ class DocumentsController < ApplicationController
   
   def index
     @application = Application.find(params[:application_id])
-    @documents = @application.documents
-  end
-
-  def new
-    @application = Application.find(params[:application_id])
-    @document = Document.new
+    @application_documents = document_lists
   end
   
-  #imtinyc@gmail.com is the public e-mail 
-  def create 
+  def create
     @application = Application.find(params[:application_id])
-    @document = Document.new(document_params)
-    if @document.save
-      flash[:success] = "Document saved."
-      finished = finish_domestic_documents(@application)
-
-      flash[:success] = "E-mail sent to confirm all neccessary documents have been uploaded." if finished
-      send_email_mailgun if finished
-
-      redirect_to application_documents_path(@application.id)
-    elsif @document.save && !finished && current_user.destroy_user_session_path
-      send_unfinished_email_mailgun
-    else
-      flash[:danger] = @document.errors.full_messages
-      render :new
-    end
+    
+    unless params[:document].nil?
+      params[:document].each do |attachment_name, attachment|
+        if @application.documents.find_by(category: attachment_name)
+          @application.documents.find_by(category: attachment_name).destroy
```

Comment:
Instead of destroying each individually, do `@application.documents.destroy_all` and then just create them